sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3994,10 +3994,45 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         c.markAsAnalyzed(AnalysisContext.get)
       case c: KeepAnalyzedQuery if c.resolved =>
         c.storeAnalyzedQuery()
-      case p: LogicalPlan if p.resolved =>
-        p.transformAllExpressionsWithPruning(_.containsPattern(ANALYSIS_AWARE_EXPRESSION)) {
-          case e: AnalysisAwareExpression[_] => e.markAsAnalyzed()
+      case c: CreateTable if c.resolved =>
+        c.copy(columns = storeDefaultForColumnDefinition(c.columns))
+      case r: ReplaceTable if r.resolved =>
+        r.copy(columns = storeDefaultForColumnDefinition(r.columns))
+      case a: AddColumns =>
```

Comment:
It is dangerous to enter this block without checking if `AddColumns` was resolved. It is all part of the analyzer. The node may not be resolved initially but will be eventually.

Reply:
yea i missed that case, anyway the code is now different, let me see if new one works